src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/SpatialCoordinateSystemManager.cs

```diff
@@ -50,9 +50,25 @@ public class SpatialCoordinateSystemManager : Singleton<SpatialCoordinateSystemM
         private readonly Dictionary<SocketEndpoint, TaskCompletionSource<bool>> remoteLocalizationSessions = new Dictionary<SocketEndpoint, TaskCompletionSource<bool>>();
         private Dictionary<SocketEndpoint, SpatialCoordinateSystemParticipant> participants = new Dictionary<SocketEndpoint, SpatialCoordinateSystemParticipant>();
         private HashSet<INetworkManager> networkManagers = new HashSet<INetworkManager>();
-        private ISpatialLocalizationSession currentLocalizationSession = null;
         private ITrackingObserver trackingObserver = null;
 
+        private readonly object localizationLock = new object();

```

Comment:
Is there a reason to prefer a separate lock object? If this is only protecting currentLocalizationSession, would we lock that object directly instead?

Reply:
locking on a null object throws exceptions. Since we want to populate currentLocalizationSession within the lock, it seems safer to lock on a separate, non-null object.